MSBuild.Sdk.Extras/build/MSBuild.Sdk.Extras.DefaultItems.targets

```diff
@@ -0,0 +1,165 @@
+<Project>
+
+	<PropertyGroup>
+		<MSBuildAllProjects>$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
+
+    <!-- _ShortFrameworkIdentifier comes from the Microsoft.NET.TargetFrameworkInference.targets -->
+    <_SdkPlatformIdentifier>$(_ShortFrameworkIdentifier)</_SdkPlatformIdentifier>
+    <!-- _SdkShortFrameworkIdentifier comes from the MSBuild.Sdk.Extras.Core.targets -->
+    <_SdkPlatformIdentifier Condition="'$(_SdkPlatformIdentifier)' == ''">$(_SdkShortFrameworkIdentifier)</_SdkPlatformIdentifier>
+
+    <!-- _TargetFrameworkVersionWithoutV comes from the Microsoft.NET.SDK -->
+    <_SdkPlatformVersionWithoutV>$(_TargetFrameworkVersionWithoutV)</_SdkPlatformVersionWithoutV>
+    <!-- _SdkTargetFrameworkVersionWithoutV comes from the MSBuild.Sdk.Extras.Core.targets -->
+    <_SdkPlatformVersionWithoutV Condition="'$(_SdkPlatformVersionWithoutV)' == ''">$(_SdkTargetFrameworkVersionWithoutV)</_SdkPlatformVersionWithoutV>
+
+    <_SdkTargetFrameworkIdentifierWithoutDot>$(TargetFrameworkIdentifier.TrimStart('.'))</_SdkTargetFrameworkIdentifierWithoutDot>
+
+    <!-- Only consider the short identifier if it doesn't match the full framework identifier already (i.e. netstandard matches) 
+         TODO: this will NOT work on case-insensitive file systems (specified on a Mac at volume creation time, but *off* by default).
+         If the file system is case sensitive and we include the wrong cased files, it won't work. 
+         In that case, users will have to opt out of the whole feature by setting EnableDefaultPlatformItems=off or by 
+         making sure the netstandard files use the proper cased TF, as in NETStandard[version].
+    -->
+    <_SdkPlatformIdentifier Condition="'$(_SdkPlatformIdentifier)' == '$(_SdkTargetFrameworkIdentifierWithoutDot)'" />
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(TargetFrameworkVersion)' != ''">
+    <_SdkTargetFrameworkMajorVersion>$(TargetFrameworkVersion.TrimStart('v'))</_SdkTargetFrameworkMajorVersion>
+    <_SdkTargetFrameworkMajorVersion>$(_SdkTargetFrameworkMajorVersion.Substring(0, $(_SdkTargetFrameworkMajorVersion.IndexOf('.'))))</_SdkTargetFrameworkMajorVersion>
+  </PropertyGroup>
+
+  <!-- Allows Platforms/[TFI]/**/*.cs, [TFI]/**/*.cs and *.[TFI].cs -->
+  <ItemGroup Label="TFI" Condition="'$(EnableDefaultItems)' == 'true' and '$(TargetFrameworkIdentifier)' != ''">
+    <!-- Switch from None > Compile for the matching framework -->    
+
+    <None Remove="Platforms/$(_SdkTargetFrameworkIdentifierWithoutDot)/**/*$(_SdkLanguageExtension)" Condition="'$(EnableDefaultNoneItems)' == 'true'" />
+    <Compile Include="Platforms/$(_SdkTargetFrameworkIdentifierWithoutDot)/**/*$(_SdkLanguageExtension)" Condition="'$(EnableDefaultCompileItems)' == 'true'" Source="$(MSBuildThisFileName)$(MSBuildThisFileExtension) (TFI)" />
```

Comment:
How would this work for projects that have multiple versions of the same platform? Many projects have `net45;net46;net471;netstandard1.0;netstandard1.3;netstandard2.0;netcoreapp2.0;netcoreapp2.1`

Just want to make sure we can handle that correctly.

Reply:
that's where the [other imports](https://github.com/kzu/MSBuildSdkExtras/blob/4ef023c5c0a3c93fdc96eaf9df040548618100ca/MSBuild.Sdk.Extras/build/MSBuild.Sdk.Extras.DefaultItems.targets#L55-L56) would kick in.

You would just define whichever is the most specific or generic according to your needs. 

For example: `Platforms\NETFramework` would be compiled for all .NET versions, `Platforms\net45` would [be compiled just for net45 short framework identifier + version](https://github.com/kzu/MSBuildSdkExtras/blob/4ef023c5c0a3c93fdc96eaf9df040548618100ca/MSBuild.Sdk.Extras/build/MSBuild.Sdk.Extras.DefaultItems.targets#L93-L94), and `Platforms\netstandard1` would build for [all versions of netstandard 1.x](https://github.com/kzu/MSBuildSdkExtras/blob/4ef023c5c0a3c93fdc96eaf9df040548618100ca/MSBuild.Sdk.Extras/build/MSBuild.Sdk.Extras.DefaultItems.targets#L74-L75) while `Platforms\netstandard2` would compile for ns2+, etc.

Reply:
maybe I should add support OOB for aliasing `NETFramework` to `net`?

Reply:
So would that work to have `Platforms\netstandard1` and `Platforms\netstandard14`?

Reply:
(I just got back today from two weeks of travel, so need to look closer as there's a lot going on here...)